shell/app-shell/app-shell.js

```diff
@@ -650,6 +659,23 @@ class AppShell extends Xen.Base {
   _onMetadata(e, metadata) {
     this._setIfDirty({metadata});
   }
+  _onShareState(e, shareState) {
+    this._setState({shareState});
+    switch (shareState) {
+      case 0:
```

Comment:
Maybe an enum instead of `0`, `1`, `2`? It looks like they're definable with `Object.freeze()` (although I don't know how much in the JS vernacular that is).

Reply:
I messed around with some enums here but am only somewhat happy with the result: https://github.com/shaper/arcs-stories/blob/master/words/source/TileBoard.js#L248

Reply:
Numerals are the original enumeration! I would probably just make constants, but I'd rather wait until the fi is higher (first thing would be factoring this code into an encapsulation anyway).